src/plugins/console/public/application/components/variables/variables_editor.tsx

```diff
@@ -0,0 +1,254 @@
+    },
+    {
+      field: 'id',
+      name: '',
+      width: '40px',
+      isExpander: true,
+      render: (id: string, variable: DevToolsVariable) => {
+        const itemIdToExpandedRowMapValues = { ...itemIdToExpandedRowMap };
+
+        return (
+          <EuiButtonIcon
+            iconType={itemIdToExpandedRowMapValues[id] ? 'arrowUp' : 'pencil'}
+            aria-label={i18n.translate('console.variablesPage.variablesTable.columns.editButton', {
+              defaultMessage: 'Edit {variable}',
+              values: { variable: variable.name },
+            })}
+            color="primary"
+            onClick={() => toggleDetails(id)}
+            data-test-subj="variableEditButton"
+          />
+        );
+      },
+    },
+    {
+      field: 'id',
+      name: '',
+      width: '40px',
+      render: (id: string, variable: DevToolsVariable) => (
+        <EuiButtonIcon
+          iconType="trash"
+          aria-label={i18n.translate('console.variablesPage.variablesTable.columns.deleteButton', {
+            defaultMessage: 'Delete {variable}',
+            values: { variable: variable.name },
+          })}
+          color="danger"
+          onClick={() => setDeleteModalForVariable(id)}
+          data-test-subj="variablesRemoveButton"
+        />
+      ),
+    },
+  ];
+
+  return (
+    <>
+      <EuiTitle>
+        <h2>
+          <FormattedMessage id="console.variablesPage.pageTitle" defaultMessage="Variables" />
+        </h2>
+      </EuiTitle>
+      <EuiSpacer size="s" />
+      <EuiText color="subdued">
+        <p>
+          <FormattedMessage
+            id="console.variablesPage.pageDescription"
+            defaultMessage="Define and reuse placeholders for dynamic values in your queries."
+          />
+        </p>
+      </EuiText>
+      <EuiSpacer size="l" />
+
+      <EuiBasicTable
+        items={variables}
+        columns={columns}
+        itemId="id"
+        responsiveBreakpoint={false}
+        className="conVariablesTable"
+        data-test-subj="variablesTable"
+        itemIdToExpandedRowMap={itemIdToExpandedRowMap}
+        noItemsMessage={i18n.translate('console.variablesPage.table.noItemsMessage', {
+          defaultMessage: 'No variables have been added yet',
+        })}
+      />
+
+      {isAddingVariable && (
+        <VariableEditorForm onSubmit={onAddVariable} onCancel={() => setIsAddingVariable(false)} />
+      )}
+
+      <EuiSpacer size="m" />
+
+      <div>
+        <EuiButton
+          data-test-subj="variablesAddButton"
+          iconType="plusInCircle"
+          onClick={() => {
+            setIsAddingVariable(true);
+            collapseExpandedRows();
+          }}
+          disabled={isAddingVariable}
+        >
+          <FormattedMessage
+            id="console.variablesPage.addButtonLabel"
+            defaultMessage="Add variable"
+          />
+        </EuiButton>
+      </div>
+
+      {deleteModalForVariable && (
```

Comment:
Hey @leemthompo, I was wondering what you thought about the copy on this modal? I'm not too sure about any of it 😬